src/linker/Linker/DependencyInfo.cs

```diff
@@ -0,0 +1,142 @@
+
+	// Generic type relationships
+		GenericArgumentType, // generic instance -> argument type
+		GenericParameterConstraintType, // generic typedef/methoddef -> parameter constraint type
+		DefaultCtorForNewConstrainedGenericArgument, // generic instance -> argument ctor
+		ElementType, // generic type instantiation -> generic typedef
+		ElementMethod, // generic method instantiation -> generic methoddef
+		ModifierType, // modified type -> type modifier
+
+	// Modules and assemblies
+		ScopeOfType, // type -> module/assembly
+		TypeInAssembly, // assembly -> type
+		ModuleOfExportedType, // exported type -> module
+		ExportedType, // type -> exported type
+
+	// Grouping of property/event methods
+		PropertyOfPropertyMethod, // property method -> its property
+		EventOfEventMethod, // event method -> its event
+		EventMethod, // event -> its event methods
+		// PropertyMethod doesn't exist because property methods aren't always marked for a property
+
+	// Interface implementations
+		InterfaceImplementationInterfaceType, // interfaceimpl -> interface type
+		InterfaceImplementationOnType, // type -> interfaceimpl on it
+
+	// Interop methods
+		ReturnTypeMarshalSpec, // interop method -> marshal spec of its return type
+		ParameterMarshalSpec, // interop method -> marshal spec of its parameters
+		FieldMarshalSpec, // field -> its marshal spec
+		InteropMethodDependency, // interop method -> required members of its parameters, return type, declaring type
+
+	// Dependencies created by instructions
+		DirectCall, // method -> method
+		VirtualCall, // method -> method
+		Ldvirtftn, // method -> method
+		Ldftn, // method -> method
+		Newobj, // method -> method
+		Ldtoken, // method -> member referenced
+		FieldAccess, // method -> field (for instructions that load/store fields)
+		InstructionTypeRef, // other instructions that have an inline type token (method -> type)
+
+	// Custom attributes on various providers
+		CustomAttribute,
+		ParameterAttribute,
+		ReturnTypeAttribute,
+		GenericParameterCustomAttribute,
+		GenericParameterConstraintCustomAttribute,
+
+	// Dependencies of custom attributes
+		AttributeConstructor, // attribute -> its ctor
+		// used for security attributes, where we mark the type/properties directly
+		AttributeType, // attribute -> attribute type
+		AttributeProperty, // attribute -> attribute ptroperty
+		CustomAttributeArgumentType,
+		CustomAttributeArgumentValue,
+		CustomAttributeField,
+
+	// Tracking cctors
+		TriggersCctorThroughFieldAccess, // field-accessing method -> cctor of field's declaring type
+		TriggersCctorForCalledMethod, // caller method -> callee method type cctor
+		DeclaringTypeOfCalledMethod, // called method -> its declaring type (used to track when a cctor is triggered by a method call)
+		CctorForType, // type -> cctor of type
+		CctorForField, // field -> cctor of field's declaring type
+
+	// Tracking instantiations
+		InstantiatedByCtor, // ctor -> type it instantiates			
```

Comment:
typos.  I think you meant

`type is instantiated`

Reply:
Updated the wording to make it more clear.